### PR TITLE
BCDA-5838: Update go dependencies / FHIR library 

### PR DESCRIPTION
**Pull request by ian-a-sawyer**
### Proposed Changes
- Updates go.mod, go.sum to latest fhir go library
- Updates corresponding fhir library code to current practice

### Security Implications

- [ ] new software dependencies

- [ ] security controls or supporting software altered

- [ ] new data stored or transmitted

- [ ] security checklist is completed for this change

- [ ] requires more information or team discussion to evaluate security implications

- [x] no PHI/PII is affected by this change

### Acceptance Validation
- Builds properly
- Unit tests pass

### CODE REVIEW

**Review by 4ell0 — DISMISSED** [review]
LGTM. Build and unit tests pass.

**Review by krobertson3314 — DISMISSED** [review]

**Review by krobertson3314 — APPROVED** [review]

File: bcda/client/bluebutton.go [hunk 1]
```diff
@@ -89,7 +89,7 @@ func NewBlueButtonClient(config BlueButtonConfig) (*BlueButtonClient, error) {
 		return nil, errors.Wrap(err, "could not load Blue Button keypair")
 	}
 
-	tlsConfig := &tls.Config{Certificates: []tls.Certificate{cert}, MinVersion: tls.VersionTLS12}
+	tlsConfig := &tls.Config{Certificates: []tls.Certificate{cert}, MinVersion: tls.VersionTLS12, NameToCertificate: nil}
```
Comment (ian-a-sawyer) [inline bcda/client/bluebutton.go:92]: This was due to a deprecation notice in the build pipeline. May you please verify this is correct?

**Review by 4ell0 — APPROVED** [review]
LGTM

**Review by magile32 — APPROVED** [review]
Soft deployment of the branch to the dev environment was succesful and verify that all automated tests passing.

**Pull request merged**